### PR TITLE
fix(redux): prevent route actions from being dispatched on hashchange

### PR DESCRIPTION
**Pull request by herschel666**
<details>
<summary>Bors merge bot cheat sheet</summary>

We are using [bors-ng](https://github.com/bors-ng/bors-ng) to automate merging of our pull requests. The following table provides a summary of commands that are available to reviewers (members of this repository with push access) and delegates (in case of `bors delegate+` or `bors delegate=[list]`).

| Syntax | Description |
| --- | --- |
| bors merge | Run the test suite and push to master if it passes. Short for "reviewed: looks good." |
| bors merge- | Cancel an r+, r=, merge, or merge= |
| bors try | Run the test suite without pushing to master. |
| bors try- | Cancel a try |
| bors delegate+ | Allow the pull request author to merge their changes. |
| bors delegate=[list] | Allow the listed users to r+ this pull request's changes. |
| bors retry | Run the previous command a second time. |

This is a short collection of opinionated commands. For a full list of the commands read the [bors reference](https://bors.tech/documentation/).

</details>

### CODE REVIEW

**Comment by KnisterPeter** [comment]
@herschel666 Can you please elaborate on why you did this change? I think its not clear by just reading the code change.

**Comment by herschel666** [comment]
@KnisterPeter The change prevents route actions from being dispatched when just the hash part of the location changes. As it is e.g. the case with jump anchor navigations.

File: packages/redux/action-creator-dispatcher/mixin.runtime-common.js [hunk 1]
```diff
@@ -14,14 +14,22 @@ const Dispatcher = withRouter(
...
+    previousPathname = null;
 
     dispatchAll() {
-      if (this.props.location) {
+      const pathnameChanged =
+        this.props.location &&
+        this.props.location.pathname !== this.previousPathname;
+
+      if (pathnameChanged) {
+        this.previousPathname = this.props.location.pathname;
```
Comment (jhiode) [inline packages/redux/action-creator-dispatcher/mixin.runtime-common.js]: Seeing this `this.props.location` so many times and also the need for truthy check lets me wonder if we shouldn't make it a little bit simpler by storing the value in a variable:

```suggestion
      const { pathname: currentPathname } = this.props.location || {};

      if (currentPathname !== this.previousPathname) {
        this.previousPathname = pathnameChanged;
```

**Comment by herschel666** [comment]
bors merge

**Pull request merged**